llvm/lib/ProfileData/MemProfReader.cpp

```diff
@@ -166,6 +166,39 @@ readMemInfoBlocksV4(const char *Ptr) {
   return Items;
 }
 
+llvm::SmallVector<std::pair<uint64_t, MemInfoBlock>>
```

Comment:
It seems this is mostly identical to the v4 reader above, with the exception of a few comments that differ (unnecessarily? makes it more difficult to see the functionality diffs), and the histogram reading loop body. Maybe refactor to make a common v4/v5 reader with just the inner part of the histogram reading loop doing different things based on version?

Reply:
Done.